### ISSUE TITLE
Count with filter generates subquery in select list

### ISSUE THREAD

**Issue by MaceWindu** [linq2db/linq2db#680]
for details and test code see #679

Example of generated query:
```sql
DataConnection: -- SqlServer.2014 SqlServer.2012
DECLARE @weekStart DateTime2
SET     @weekStart = '2017-05-22'
DECLARE @monthStart DateTime2
SET     @monthStart = '2017-05-01'
DECLARE @p1 DateTime2
SET     @p1 = '2017-02-23T20:31:57'
DECLARE @yearStart DateTime2
SET     @yearStart = '2017-01-01'
DECLARE @p2 DateTime2
SET     @p2 = '2016-11-24T20:31:57'

SELECT TOP (1)
	(
		SELECT
			Count(*)
		FROM
			[Database].[Schema].[Table] [keyParam]
		WHERE
			[t1].[SentTimestamp] = [keyParam].[SentTimestamp] AND
			[keyParam].[SentTimestamp] > @weekStart
	) as [c1],
	(
		SELECT
			Count(*)
		FROM
			[Database].[Schema].[Table] [keyParam1]
		WHERE
			[t1].[SentTimestamp] = [keyParam1].[SentTimestamp] AND
			[keyParam1].[SentTimestamp] > @monthStart
	) as [c2],
	(
		SELECT
			Count(*)
		FROM
			[Database].[Schema].[Table] [keyParam2]
		WHERE
			[t1].[SentTimestamp] = [keyParam2].[SentTimestamp] AND
			[keyParam2].[SentTimestamp] > @p1
	) as [c3],
	(
		SELECT
			Count(*)
		FROM
			[Database].[Schema].[Table] [keyParam3]
		WHERE
			[t1].[SentTimestamp] = [keyParam3].[SentTimestamp] AND
			[keyParam3].[SentTimestamp] > @yearStart
	) as [c4],
	(
		SELECT
			Count(*)
		FROM
			[Database].[Schema].[Table] [keyParam4]
		WHERE
			[t1].[SentTimestamp] = [keyParam4].[SentTimestamp] AND
			[keyParam4].[SentTimestamp] > @p2
	) as [c5]
FROM
	[Database].[Schema].[Table] [t1]
GROUP BY
	[t1].[SentTimestamp]

```

**Comment by ili** [comment]
@MaceWindu can you please make branch & PR with test reproducing the issue?

**Comment by MaceWindu** [comment]
ok

**Comment by sdanyliv** [comment]
@MaceWindu, looks like it is already fixed in #679.
Yes there is new function for that `CountExt` but it works.
